S 1 security token service

andreyryzhkov opened the pull request.

andreyryzhkov committed:
[s-1] Security token service (initial commit)

1) Added UserDao
2) Created entities: User, Credential, Session, SessionDto
3) Added SecurityService
4) Added Login Controller

andreyryzhkov committed:
[s-1] Security token service

1) Added Logout
2) Re-factor cache

Rooman commented on src/main/java/com/aryzhkov/movieland/security/impl/SecurityServiceImpl.java:
```diff
@@ -0,0 +1,61 @@
+@RequiredArgsConstructor
+public class SecurityServiceImpl implements SecurityService {
+
+    @Value("2")
```
@Value("${session.timeout:2}")

Rooman commented on src/main/java/com/aryzhkov/movieland/dao/jdbc/JdbcUserDao.java:
```diff
@@ -0,0 +1,26 @@
+public class JdbcUserDao implements UserDao {
+
+    private static final String SELECT_BY_EMAIL = "select user_id, first_name || ' ' || last_name as nick_name" +
+            "  from movieland.users where email = ? and password = ?";
```
please add BCrypt

andreyryzhkov committed:
[s-1] Security token service

1) Fixed after review
2) Change logic for getSession
3) Moved isSessionExpired to SecurityService
4) Moved value session expired time to app property

andreyryzhkov committed:
[s-1] Security token service

1) Fixed after review
2) Added crypto password
3) Added tests

Rooman commented on src/main/java/com/aryzhkov/movieland/dao/jdbc/JdbcUserDao.java:
```diff
@@ -0,0 +1,34 @@
+    private final JdbcTemplate jdbcTemplate;
+
+    @Override
+    public User getByEmail(Credential credential) {
```
Please use Optional here
